### PR TITLE
Adding test (WIP)

### PR DESCRIPTION
**Pull request by aheficent**

### CODE REVIEW

File: bi_sql_editor/models/bi_sql_view.py [hunk 1]
```diff
@@ -143,12 +143,13 @@ def onchange_group_ids(self):
     # Overload Section
     @api.multi
     def unlink(self):
+        self.ensure_one()
```
Comment (legalsylvain) [inline bi_sql_editor/models/bi_sql_view.py]: why ? We are loosing the possiblitity to unlink many draft item in one time. (tree view button).

File: bi_sql_editor/models/bi_sql_view.py [hunk 2]
```diff
@@ -143,12 +143,13 @@ def onchange_group_ids(self):
...
         non_draft_views = self.search([
             ('id', 'in', self.ids),
             ('state', 'not in', ('draft', 'sql_valid'))])
         if non_draft_views:
             raise UserError(_("You can only unlink draft views"))
-        return self.unlink()
+        return super(BiSQLView, self).unlink()
```
Comment (legalsylvain) [inline bi_sql_editor/models/bi_sql_view.py]: Thanks a lot ! My apologize for this trivial bug.

File: bi_sql_editor/tests/test_bi_sql_view.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+
+class TestBiSqlViewEditor(TransactionCase):
+
+    def setUp(self):
+        super(TestBiSqlViewEditor, self).setUp()
+        self.res_partner = self.env['res.partner']
+        self.res_users= self.env['res.users']
+        self.bi_sql_view = self.env['bi.sql.view']
+        self.view = self.create_view('partner_agenda', 'agenda_view',
```
Comment (legalsylvain) [inline bi_sql_editor/tests/test_bi_sql_view.py]: You could maybe drop this part, and create 'partner_agenda' by demo data (in a draft state). So, users can use this demo data to test the module, and you can too use it in the Test part. don't you think ?

File: bi_sql_editor/tests/test_bi_sql_view.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        return view
+
+    def test_process_view(self):
+        self.view.button_validate_sql_expression()
+        self.view._check_execution()
+        for field in self.view.bi_sql_view_field_ids:
+            field.graph_type = 'row'
+        self.view.button_create_sql_view_and_model()
+        self.view.button_create_ui()
+        self.view.button_create_ui()
```
Comment (legalsylvain) [inline bi_sql_editor/tests/test_bi_sql_view.py]: Is it wanted to have written two times the same line ?

File: bi_sql_editor/models/bi_sql_view.py (hunk 1 above)
Comment (aheficent) [inline bi_sql_editor/models/bi_sql_view.py]: yes this was a mistake. I'll remove this

File: bi_sql_editor/tests/test_bi_sql_view.py (hunk 4 above)
Comment (aheficent) [inline bi_sql_editor/tests/test_bi_sql_view.py]: Not really. It's  a mistake. Thanks

File: bi_sql_editor/tests/test_bi_sql_view.py (hunk 3 above)
Comment (aheficent) [inline bi_sql_editor/tests/test_bi_sql_view.py]: that's better

**Comment by aheficent** [comment]
Changes are done, but I cannot reopen this. Should I create a new one?

**Comment by legalsylvain** [comment]
you're making a PR against adhoc-dev repository. OCA members have no right to reopen a PR. You should ask to @nicomacr.

**Pull request closed without merging**